### PR TITLE
planner: fix order by sub-query couldn't find outer correlated columns

### LINKED ISSUE #26945: Table alias is not propagated to subquery
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
DROP TABLE IF EXISTS t1, t2;
CREATE TABLE t1 (a INT, b INT);
CREATE TABLE t2 (a INT, b INT);
INSERT INTO t1 VALUES (1, 1);
INSERT INTO t2 VALUES (1, 1);
SELECT one.a FROM t1 one ORDER BY (SELECT two.b FROM t2 two WHERE two.a = one.b);
```

### 2. What did you expect to see? (Required)

```
+------+
| a    |
+------+
|    1 |
+------+
1 row in set (0.00 sec)

```

### 3. What did you see instead (Required)

```
ERROR 1054 (42S22): Unknown column 'one.b' in 'where clause'
```

### 4. What is your TiDB version? (Required)

```
Release Version: v5.1.0
Edition: Community
Git Commit Hash: 8acd5c88471cb7b4d4c4a8ed73b4d53d6833f13e
Git Branch: heads/refs/tags/v5.1.0
UTC Build Time: 2021-06-24 07:10:32
GoVersion: go1.16.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: close #26945

Problem Summary:
```
DROP TABLE IF EXISTS t1, t2;
CREATE TABLE t1 (a INT, b INT);
CREATE TABLE t2 (a INT, b INT);
INSERT INTO t1 VALUES (1, 1);
INSERT INTO t2 VALUES (1, 1);
SELECT one.a FROM t1 one ORDER BY (SELECT two.b FROM t2 two WHERE two.a = one.b);
```

### What is changed and how it works?
fetch the correlated column in order-by subquery out to select fields, building it as an Auxiliary one, which will be stored as p.schema() (passing as outer schema when enter subquery) when we try to build order-by clause. Otherwise it couldn't find that correlated column.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -3758,6 +3800,10 @@ func (b *PlanBuilder) buildSelect(ctx context.Context, sel *ast.SelectStmt) (p L
 		}
 	}
 
+	if strings.HasPrefix(b.ctx.GetSessionVars().StmtCtx.OriginalSQL, "select * from (select 1 as a") {
+		fmt.Println(1)
+	}
```
Comment (winoros) [inline planner/core/logical_plan_builder.go:3805]: remove these lines

File: planner/core/logical_plan_builder.go [hunk 2]
```diff
@@ -2219,6 +2219,48 @@ func (b *PlanBuilder) resolveHavingAndOrderBy(sel *ast.SelectStmt, p LogicalPlan
...
+							sel.Fields.Fields = append(sel.Fields.Fields, &ast.SelectField{
+								Auxiliary: true,
+								Expr:      columnNameExpr,
+							})
+						}
+					}
+				}
+			}
+		}
+	}
```
Comment (winoros) [inline planner/core/logical_plan_builder.go:2263]: Add some comments for this code block.

**Review by winoros — COMMENTED** [review]
Than main changes LGTM

**Review by winoros — APPROVED** [review]

**Comment by AilinKid** [comment]
This PR has come problems with CLA since I migrate to new mbp recently with wrong-set git configuration.
It's has redirected to https://github.com/pingcap/tidb/pull/33640

**Review by xuyifangreeneyes — APPROVED** [review]

**Pull request closed without merging**